gnucash/gnome/gnc-plugin-page-report.cpp

```diff
@@ -1162,14 +1157,13 @@ gnc_plugin_page_report_constructor(GType this_type, guint n_properties, GObjectC
     GncPluginPageReportClass *our_class;
     GObjectClass *parent_class;
     gint reportId = -42;
-    int i;
 
     our_class = GNC_PLUGIN_PAGE_REPORT_CLASS (
                     g_type_class_peek (GNC_TYPE_PLUGIN_PAGE_REPORT));
     parent_class = G_OBJECT_CLASS (g_type_class_peek_parent (our_class));
     obj = parent_class->constructor(this_type, n_properties, properties);
 
-    for (i = 0; i < n_properties; i++)
+    for (decltype(n_properties) i = 0; i < n_properties; i++)
```

Comment:
Just for my understanding - What does `decltype(n_propterties)` add here ? Will it ensure variable `i` is of the same type as `n_properties` ?

Reply:
Yes, `decltype(n_properties)` is the type of i. If I'd said `auto i` instead i would be an int and some compilers would complain about the type mismatch in `i < n_properites'.

Reply:
This is neat. In some of these situations, I have used `static_cast`. Using `decltype` pattern expresses intent better. Thanks @jralls

Reply:
Agreed this is quite a neat new feature.